Updated dependencies and added auto id3 tagging

matteo05vava opened the pull request:
Hi,
i've updated all the dependencies to the latest and also added an auto-retrieve for the id3 tags (if present) for songs.

matteo05vava committed:
Update dependencies

matteo05vava committed:
Update youtube-ytdl.js promises

matteo05vava committed:
Update version

matteo05vava committed:
Update package-lock.json

matteo05vava committed:
Update id3 to write album, song and artist if present

maxill1 commented:
merged, set up a pipeline ad published on npmjs as 0.0.6 (I published 0.0.5 months ago https://github.com/maxill1/node-red-contrib-ytdl/commit/a459427f49aaac996066902e2c66c5e24168a65a, but forgot tu push package.json... sorry). 

I haven't tested, let me know if it's fine.

matteo05vava commented:
> merged, set up a pipeline ad published on npmjs as 0.0.6 (I published 0.0.5 months ago [a459427](https://github.com/maxill1/node-red-contrib-ytdl/commit/a459427f49aaac996066902e2c66c5e24168a65a), but forgot tu push package.json... sorry).
> 
> I haven't tested, let me know if it's fine.

Ok, no problem!

I've updated all the dependencies because the node was not working anymore (it returns a "configuration error for youtube").
After the dependencies update both ytpl and ytdl-core return an unhandled promise for the "info" callback, so i've changed the structure of the function in order to use ".then" and ".catch" instead of ".function (info, err)".

I've changed the title.replace with some regex in order to change all chars and not only the first one. i've a doubt on the row 30 <<title = title.replace(/\\/g, "-");>> because here i replace every single "\" with a "-", but i've seen that in your original code you was replacing only the double backslash "\\" with a "-", if you want to replace only if there is the double backlash you need to change the regex to: "/\\\\/g".

Moreover i've used the youtube media details to retrieve and write id3 tags if present on the video.

I've tested it with a couple of playlist, some single videos and arrays of videos and works, i don't know if you have some tests to do.

proton555 commented:
@maxill1 @matteo05vava 
I am sorry as i had no other option to open an issue/contact you

Can you kindly give some details on how to customize downloads? Currently only one size/format video file this node downloads, is also without any metadata and not even accepted/played by many video players. I checked the ytdlcore and it seems this node uses a custom filter but more importantly does not pass the "info" param separately(?) 
This README https://github.com/fent/node-ytdl-core/blob/master/README.md says
` "Supplying this option will ignore the filter and quality options since the format is explicitly provided."`

Can you please kindly :
1. add option to choose formats than 1 custom one, so that different qualities can be chosen
2. fix metadata or mediainfo missing on final downloaded video files
3. Also it seems extension of files are saved in caps

Or kindly give some hint on fixing them especially 2nd problem. 
There is no other working node about it, so kindly help. Thanks
